projects/pgai/tests/vectorizer/extensions/fixtures/migrations/002_create_vectorizer.py.template

```diff
@@ -0,0 +1,49 @@
+"""create vectorizer
+
+Revision ID: {revision_id}
+Revises: {revises}
+Create Date: {create_date}
+"""
+from alembic import op
+from pgai.vectorizer.configuration import (
+   OpenAIEmbeddingConfig,
+   CharacterTextSplitterConfig,
+   PythonTemplateConfig
+)
+from sqlalchemy import text
+
+# revision identifiers
+revision = '{revision_id}'
+down_revision = '{down_revision}'
+branch_labels = None
+depends_on = None
+
+def upgrade():
+    op.create_vectorizer(
+        'blog',
+        target_table='blog_embeddings',
+        embedding=OpenAIEmbeddingConfig(
+            model='text-embedding-3-small',
+            dimensions=768
+        ),
+        chunking=CharacterTextSplitterConfig(
+            chunk_column='content',
+            chunk_size=800,
+            chunk_overlap=400,
+            separator='.',
+            is_separator_regex=False
+        ),
```

Comment:
This is a bit of an ugly side effect of having merged the parent classes now.
The pydantic models that map the db config have these fields as non optional because the extension code provides the defaults.

But the **input function** should have these as optional for the same reason (if they are null the extension will provide e.g. `chunk_size=800`. 

So now in the python code you have to unnecessarily always provide all the arguments (true for some of the other configs as well).

Alternatively I could give these the same defaults as the extension code provides, but then we have to again maintain the defaults in 2 places.

This was previously not an issue as I could just make the input arguments nullable while leaving the load from db models required. I can override this of course in the child classes, but then again this breaks the purpose of sharing a parent class in the first place.